scripts/gpg-share.sh

```diff
@@ -125,36 +108,105 @@ check_users() {
 }
 
 
-check_users
-
-DUMP_DIR=$(mktemp -u "/tmp/gpg-share-XXXXXX")
-mkdir -p "$DUMP_DIR"
-
-
 # TODO: documentation
 encrypt () {
-  for _user in "${users[@]}";
+  file="$1"
+  directory="$2"
+  shift 2
+  users=("$@")
+  for user in "${users[@]}";
   do
-      user=$(echo -n "$_user")
-      log_info "encrypting '$file' for '$user' inside '$DUMP_DIR'..."
-      output=$(echo "$DUMP_DIR/$file.$user.asc" | sed 's/\s\+/-/g')
+      log_info "encrypting '$file' for '$user' inside '$directory'..."
+      output=$(echo "$directory/$file.$user.asc" | sed 's/\s\+/-/g')
       gpg --verbose --recipient "$user" --encrypt --armor --output "$output" "$file"
   done
-  log_success "all encrypted file have been saved inside '$DUMP_DIR'"
+  log_success "all encrypted file have been saved inside '$directory'"
 }
 
 
 # TODO: documentation
 archive () {
+  directory="$1"
+  name="$2"
   log_info "storing files in archive"
   # subshell here to avoid archiving the paths to the .asc files
   (
-    cd "$DUMP_DIR"
-    tar cvf "$ARCHIVE" $(find . -type f)
+    cd "$directory"
+    tar cvf "$name" $(find . -type f)
   )
-  cp "$DUMP_DIR/$ARCHIVE" .
+  cp "$directory/$name" .
+}
+
+
+# parse the arguments.
+OPTIONS=$(getopt -o hu:U:f: --long help,user:,users:,file -n 'gpg-share' -- "$@")
+if [ $? != 0 ] ; then echo "Terminating..." >&2 ; exit 1 ; fi
+eval set -- "$OPTIONS"
+
+
+usage () {
+  echo "Usage: $USAGE"
+  echo "Type -h or --help for the full help."
+  exit 0
+}
+
+
+help () {
+  echo "gpg-share:"
+  echo "     a script to share a secret file with others using pgp public keys."
+  echo "     it should guide you when you make mistakes, e.g. providing a non-existing file."
+  echo ""
+  echo "Usage:"
+  echo "     $USAGE"
+  echo ""
+  echo "Options:"
+  echo "     -h/--help               shows this help."
+  echo "     -f/--file               the secret file to share."
+  echo "     -u/--user               a single user id in the keyring."
+  echo "                               can be used multiple times."
+  echo "                               supports multi-word ids inside quotes, e.g. \"John Smith\"."
+  echo "     -U/--users              the path to a file containing pgp ids."
+  echo "                               should contain exactly one valid pgp id per line."
+  echo "                               can be used multiple times."
+  exit 0
+}
+
+
+main () {
+  users=()
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      -h | --help ) help ;;
+      -f | --file ) file="$2"; shift 2;;
```

Comment:
maybe add a check to forbid multiple use of `-f`? (which would end up not stacking but overwriting previous arg)

Reply:
not sure i got what you propose :thinking: 

- do you want to **throw an error when `-f` is used multiple times**?
- do you want to allow the use of multiple `-f`s and do as with the users and **build an array of files to encrypt and share**?

Reply:
I would like to **forbid** it so throwing an error seems good :+1:

Reply:
so we'll encrypt a single file then :wink:

Reply:
fix available in 86de353